builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -65,6 +65,13 @@ func resourceAwsDbInstance() *schema.Resource {
 				Computed: true,
 			},
 
+			"character_set_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```

Comment:
If you wan't to change the character set, then you must create a new resource. So it's an input a creation time, and if you decide you want to change it, then the database *must* be recreated - the old destroyed and a new created.